agent/cmd/edgemesh-agent/app/options/options.go

```diff
@@ -43,5 +43,6 @@ func (o *EdgeMeshAgentOptions) Config() (*config.EdgeMeshAgentConfig, error) {
 	if err := cfg.Parse(o.ConfigFile); err != nil {
 		return nil, err
 	}
+	cfg.Modules.TunnelAgentConfig.IsGatewayEnable = cfg.Modules.EdgeGatewayConfig.Enable
```

Comment:
Not a good position, I think.
It would be better to put this code in `agent/cmd/edgemesh-agent/app/server.go` and add your comments, like:
```
// registerModules register all the modules started in edgemesh-agent
func registerModules(c *config.EdgeMeshAgentConfig, ifm *informers.Manager) []error {
	var errs []error
	if err := dns.Register(c.Modules.EdgeDNSConfig, ifm); err != nil {
		errs = append(errs, err)
	}
	if err := proxy.Register(c.Modules.EdgeProxyConfig, ifm); err != nil {
		errs = append(errs, err)
	}
	if err := gateway.Register(c.Modules.EdgeGatewayConfig, ifm); err != nil {
		errs = append(errs, err)
	}

	// TODO add your comments
	// Edit your code
	cfg.Modules.TunnelAgentConfig.IsGatewayEnable = cfg.Modules.EdgeGatewayConfig.Enable
	if err := tunnel.Register(c.Modules.TunnelAgentConfig, ifm); err != nil {
		errs = append(errs, err)
	}
	return errs
}
```

Reply:
ok, i will delete this config element and add tunnelMode to control the tunnel behavior